src/runtime/node/timers/internal/immediate.ts

```diff
@@ -17,3 +17,31 @@ export class Immediate<TArgs extends any[]> implements NodeJS.Immediate {
   }
   [Symbol.dispose]() {}
 }
+
+export class ImmediateTimeoutBased<TArgs extends any[]>
+  implements NodeJS.Immediate
+{
+  _onImmediate: (...args: TArgs) => void;
+  private _timeout: NodeJS.Timeout;
+
+  constructor(callback: (...args: TArgs) => void, args: TArgs) {
+    this._onImmediate = callback;
+    this._timeout = setTimeout(callback, 0, ...args);
```

Comment:
I think you can directly update `Immediate` class implementation to make use of timeout (and we can extract this change as a separate PR)

Reply:
So the change would not be cf specific?

Reply:
Yes! Using setTimeout seems obviously better choice than immediately calling fn